consensus/istanbul/core/persistent_state.go

```diff
@@ -0,0 +1,96 @@
+package core
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/log"
+	"io/ioutil"
+	"math/big"
+	"os"
+	"path/filepath"
+)
+
+func (c *core) savePrepareMessageToDisk(
+	messageType uint64,
+	roundNumber *big.Int,
+	sequenceNumber *big.Int,
+	msg []byte) error {
+	fileName, err := c.generateFileName(messageType, roundNumber, sequenceNumber)
+	if err != nil {
+		return err
+	}
+	err2 := writeToDisk(fileName, msg)
+	log.Debug("savePrepareMessageToDisk/wrote file to the disk", "file", fileName, "error", err2)
+	return err2
+}
+
+// getPreprepareMessageFromDisk returns the prepared message
+// If the file does not exist, it returns (nil, nil).
+// It returns an error for all other failure cases.
+func (c *core) getPreprepareMessageFromDisk(
+	messageType uint64,
+	roundNumber *big.Int,
+	sequenceNumber *big.Int) ([]byte, error) {
+	fileName, err := c.generateFileName(messageType, roundNumber, sequenceNumber)
+	if err != nil {
+		return nil, err
+	}
+	data, err := ioutil.ReadFile(fileName)
+	if os.IsNotExist(err) {
+		log.Debug("getPreprepareMessageFromDisk/file does not exist", "file", fileName)
+		return nil, nil
+	}
+	log.Debug("getPreprepareMessageFromDisk/file found on the disk", "file", fileName)
+	return data, err
+}
+
+// deletePrepareMessageFromDisk deletes all files from this round
+func (c *core) deletePrepareMessageFromDisk(roundNumber *big.Int) error {
+	dir := c.backend.GetDataDir()
+	// This pattern must be  similar to the filenames generated by
+	// generateFileName function.
+	filePattern := filepath.Join(dir,
+		fmt.Sprintf("geth_istanbul_state_type_.*_round_%s_.*", roundNumber.String()))
```

Comment:
we shouldn't delete files associated with (possibly subsequent) sequences all at once here.

Reply:
Changed the code to be conservative and delete only the files from current round and sequence.

Reply:
My one concern with how we delete old messages is that we need to be careful in when we call this function, likely for the range of [oldest prepare found, curr seq) on each new seq if there is a file saved here or just for the previous seq when starting a new seq.

It's the challenge of trying to match numeric equality to a regex.

Reply:
Fair point. @trianglesphere , for now, I have reduced the code to only delete files for a particular sequence number and round number. Missing deletion does not affect correctness of the code and more can be added as a follow-on.